lib/pbench/server/api/resources/users_api.py

```diff
@@ -284,24 +285,55 @@ def post(self):
                     }
         """
         auth_token = Auth.get_auth_token()
-        user = Auth.verify_auth(auth_token=auth_token)
+        state = Auth.verify_internal_token(auth_token)
 
-        # "None" user represents that either the token is not present in our database or it is an expired token.
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with the user, remove it.
 
-        if user:
-            try:
-                ActiveTokens.delete(auth_token)
-            except Exception:
-                current_app.logger.exception(
-                    "Exception occurred deleting an auth token"
-                )
-                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-            else:
-                current_app.logger.debug("User {} logged out", user.username)
+        if state == Auth.TokenState.INVALID:
+            current_app.logger.info("User logout with invalid token: {}", auth_token)
         else:
-            current_app.logger.info("User logout with invalid or expired token")
+            token = ActiveTokens.query(auth_token)
+            if token:
+                user = token.user
+                if not user:
+                    # This is an internal error state, but play it cool with the
+                    # API client.
+                    current_app.logger.error(
+                        "INTERNAL ERROR - token {!r} ({}) found without an associated user",
+                        auth_token,
+                        state,
+                    )
+                else:
+                    try:
+                        ActiveTokens.delete(auth_token)
+                    except Exception:
+                        current_app.logger.exception(
+                            "Exception occurred deleting auth token {!r} for user {!r}",
+                            auth_token,
+                            user.username,
+                        )
+                        if state == Auth.TokenState.VERIFIED:
+                            # Only report an internal error to the user if the token
+                            # has been verified.
+                            abort(
+                                HTTPStatus.INTERNAL_SERVER_ERROR,
+                                message="INTERNAL ERROR",
+                            )
```

Comment:
Why bother conditionalizing this?  At this point, it is possible that the token is valid, in the database, associated with a user, but expired...what difference does it make if it is un-expired?

We've already filtered out the case of _invalid_ tokens; I would think the rest of them would be handled the same way at this point.

Reply:
I'll have to review this more, I am finding it difficult to review the code in the way GitHub is presenting it to me at the moment.

Reply:
Given https://github.com/distributed-system-analysis/pbench/pull/3231#discussion_r1107091136, I think this can be removed.